iwyu.cc

```diff
@@ -2594,8 +2583,17 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
     vector<const UsingDecl*> using_decls
         = FindInMultiMap(visitor_state_->using_declarations, decl);
     for (Each<const UsingDecl*> it(&using_decls); !it.AtEnd(); ++it) {
-      if (!(*it)->getDeclContext()->Encloses(using_context))
-        continue;
+      const DeclContext* using_decl_context = (*it)->getDeclContext();
+      if (!using_decl_context->Encloses(use_context)) {
```

Comment:
> I think if Encloses fails you could just check if the using-decl's parent is the same as the decl's parent, and if either of them is a FunctionDecl.

I'm pretty sure I tried that as I tried a few permutations before I settled here.

> I'll ask the Clang devs if Encloses is broken in this regard, it could be that we could get it fixed.

Fantastic!